lib/pbench/server/api/resources/query_apis/__init__.py

```diff
@@ -104,12 +104,20 @@ def convert_username(value: str) -> str:
 
     Raises:
         ConversionError: input can't be converted
+        or
+        UnVerifiedUser: If the username provided does not match with the username encoded the authorization header
 
     Returns:
         internal username representation
     """
     try:
-        return Auth.validate_user(value)
+        user, verified = Auth().verify_user(value)
+        if verified or user.is_admin():
+            return value
+        else:
+            raise UnVerifiedUser(value)
+    except UnVerifiedUser:
+        raise
```

Comment:
Yeah, I failed to notice that you're returning 401, and you're right that's appropriate.

So a separate handler may be necessary; I think `SchemaError` still fits, however; although it might be a peer of the `ConversionError` subclass rather than a subclass of `ConversionError`. (It is designed to be reported by the schema processing, after all.)

Of course, since the whole Schema is designed to process the JSON parameters to queries, another option would be to expand `SchemaError` (and subclasses) with an HTTP return status; then you could just handle `SchemaError` and abort with the appropriate status.

E.g., a keyword constructor parameter, like

```
class SchemaError(Exception):
   def __init__(self, status: int = 400):
        self.http_status = status
        ...

class UnverifiedUser(SchemaError):
    def __init__(self, user: str):
        super().__init__(status = 401)
        ...
```